tests/foreman/destructive/test_infoblox.py

```diff
@@ -76,3 +110,147 @@ def test_plugin_installation(target_sat, command_args, command_opts, rpm_command
+
+    installer = module_target_sat.install(
+        InstallerCommand(help='| grep foreman-proxy-dns-provider')
+    )
+    assert 'current: "infoblox"' in installer.stdout
+
+    installer = module_target_sat.install(InstallerCommand(help='| grep foreman-proxy-dhcp-server'))
+    assert f'current: "{settings.infoblox.hostname}"' in installer.stdout
+
+    installer = module_target_sat.install(
+        InstallerCommand(help='| grep foreman-proxy-plugin-dns-infoblox-dns-server')
+    )
+    assert f'current: "{settings.infoblox.hostname}"' in installer.stdout
+
+    macaddress = gen_mac(multicast=False)
+    # using the domain name as defined in Infoblox DNS
+    domain = module_target_sat.api.Domain(
+        name=settings.infoblox.domain,
+        location=[module_location],
+        dns=module_provisioning_capsule.id,
+        organization=[module_sca_manifest_org],
+    ).create()
+    subnet = module_target_sat.api.Subnet(
+        location=[module_location],
+        organization=[module_sca_manifest_org],
+        network=settings.infoblox.network,
+        cidr=settings.infoblox.network_prefix,
+        mask=settings.infoblox.netmask,
+        from_=settings.infoblox.start_range,
+        to=settings.infoblox.end_range,
+        boot_mode='DHCP',
+        ipam='DHCP',
+        dhcp=module_provisioning_capsule.id,
+        tftp=module_provisioning_capsule.id,
+        dns=module_provisioning_capsule.id,
+        discovery=module_provisioning_capsule.id,
+        remote_execution_proxy=[module_provisioning_capsule.id],
+        domain=[domain.id],
+    ).create()
+    host = module_target_sat.api.Host(
+        organization=module_sca_manifest_org,
+        location=module_location,
+        name=gen_string('alpha').lower(),
+        mac=macaddress,
+        operatingsystem=module_sync_kickstart_content.os,
+        architecture=default_architecture,
+        domain=domain,
+        subnet=subnet,
+        root_pass=settings.provisioning.host_root_password,
+        ptable=default_partitiontable,
+        content_facet_attributes={
+            'content_source_id': module_provisioning_capsule.id,
+            'content_view_id': module_default_org_view.id,
+            'lifecycle_environment_id': module_lce_library.id,
+        },
+    ).create()
+    # check if A Record is created for the host IP on Infoblox
+    url = f'https://{settings.infoblox.hostname}/wapi/v2.0/ipv4address?ip_address={host.ip}'
+    auth = (settings.infoblox.username, settings.infoblox.password)
+    result = requests.get(url, auth=auth, verify=False)
+    assert result.status_code == 200
+    # check hostname and ip is present in A record
+    assert host.name in result.text
+    assert host.ip in result.text
+    # disable dhcp and dns plugin
+    if is_open('BZ:2210256'):
+        disable_infoblox_plugin = InstallerCommand(installer_args=infoblox_plugin_disable)
+        result = module_target_sat.install(disable_infoblox_plugin)
```

Comment:
Can we add RC/status check after this installer command to verify the installer passed?
And shouldn't we skip this disable part here if BZ is open? I think we're running into this BZ again and PRT is passing as we're only verifying if flags are set to false after it, right?

Reply:
Added RC/status check and corrected the disable part